deploy/create-bridge-token-proposal.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  Contract,
+  Interface,
+  AddressLike,
+  BigNumberish,
+  parseEther,
+  formatEther,
+} from "ethers";
+import { utils } from "zksync-ethers";
+import * as hre from "hardhat";
+
+import { getWallet, getL1Wallet, getEnvVariable } from "./utils";
+import constants from "./constants";
+
+export default async function () {
+  const wallet = getWallet();
+  const l1Wallet = getL1Wallet();
+
+  // Proposal Creation on L1
+
+  const governorArtifact = await hre.artifacts.readArtifact("IGovernor");
+  const governor = new Contract(
+    constants.addresses.L1_GOVERNOR,
+    [
+      ...governorArtifact.abi,
+      {
+        type: "function",
+        name: "queue",
+        inputs: [
+          { name: "targets", type: "address[]", internalType: "address[]" },
+          { name: "values", type: "uint256[]", internalType: "uint256[]" },
+          { name: "calldatas", type: "bytes[]", internalType: "bytes[]" },
+          { name: "descriptionHash", type: "bytes32", internalType: "bytes32" },
+        ],
+        outputs: [{ name: "", type: "uint256", internalType: "uint256" }],
+        stateMutability: "nonpayable",
+      },
+    ],
+    l1Wallet
+  );
+
+  const targets: AddressLike[] = [];
+  const values: BigNumberish[] = [];
+  const calldatas: string[] = [];
+
+  const merkleRoot = getEnvVariable("MERKLE_ROOT");
+  const ipfsHash = getEnvVariable("IPFS_HASH");
+  const tokenAmount = parseEther("400000");
+
+  /**
+   * LAYER 1
+   */
+
+  console.log(
+    "Step 1: Encode approve(L1ERC20Bridge, amount) to IncrementToken"
+  );
+  const IncrementTokenArtifact = await hre.artifacts.readArtifact("ERC20");
+  const incrementTokenInterface = new Interface(IncrementTokenArtifact.abi);
+  targets.push(constants.addresses.L1_TOKEN);
+  values.push(0);
+  calldatas.push(
+    incrementTokenInterface.encodeFunctionData("approve", [
+      constants.addresses.L1_BRIDGE,
+      tokenAmount,
+    ])
+  );
+
+  console.log("Step 2: Encode deposit to L1Bridge");
+  const l1BridgeInterface = utils.L1_BRIDGE_ABI;
+  const l1Bridge = new Contract(
+    constants.addresses.L1_BRIDGE,
+    l1BridgeInterface,
+    l1Wallet
+  );
+  const l2GasEstimate = await utils.estimateDefaultBridgeDepositL2Gas(
+    l1Wallet.provider,
+    wallet.provider,
+    constants.addresses.L1_TOKEN,
+    tokenAmount,
+    constants.addresses.OWNED_MULTICALL,
+    constants.addresses.L1_TIMELOCK
+  );
+  targets.push(constants.addresses.L1_BRIDGE);
+  values.push(0);
+  calldatas.push(
+    l1BridgeInterface.encodeFunctionData("deposit", [
+      constants.addresses.OWNED_MULTICALL,
+      constants.addresses.L1_TOKEN,
+      tokenAmount,
+      l2GasEstimate,
```

Comment:
We will probably want to double or triple this estimate to ensure the transaction is executed after the voting threshold